resources/tentacle.rb

```diff
@@ -19,28 +19,211 @@
+      .\\Tentacle.exe configure --instance="#{instance}" --home="#{home_path}" --app="#{app_path}" --port="#{port}" --noListen="#{powershell_boolean(polling)}" --console
+      #{catch_powershell_error('Configuring instance')}
+      .\\Tentacle.exe configure --instance="#{instance}" --trust="#{trusted_cert}" --console
+      #{catch_powershell_error('Trusting Octopus Deploy Server')}
+      .\\Tentacle.exe service --instance="#{instance}" --install --console
+      #{catch_powershell_error('Installing Service')}
+    EOH
+    not_if { ::Win32::Service.exists?(service_name) }
+  end
+
+  windows_firewall_rule 'Octopus Deploy Tentacle' do
+    action :create
+    localport port.to_s
+    dir :in
+    protocol 'TCP'
+    firewall_action :allow
+    only_if { configure_firewall }
+    not_if { polling }
+  end
+
+  # Make sure enabled and started
+  windows_service service_name do
+    action [:enable, :start]
+    run_as_user service_user
+    run_as_password service_password
+    sensitive new_resource.sensitive
+  end
+end
+
+action :register do
+  port = resolve_port(polling, new_resource.port)
+  environment = Array(new_resource.environment)
+  service_name = service_name(instance)
+
+  verify_server(server)
+  verify_api_key(api_key)
+  verify_roles(roles)
+  verify_environment(environment)
+
+  powershell_script "register-#{instance}-tentacle" do
+    action :run
+    cwd tentacle_install_location
+    code <<-EOH
+      .\\Tentacle.exe register-with --instance "#{instance}" --server "#{server}" --name "#{tentacle_name}" --publicHostName "#{public_dns}" --apiKey "#{api_key}" #{register_comm_config(polling, port)} #{option_list('environment', environment)} #{option_list('role', roles)} #{option_list('tenant', tenants)} #{option_list('tenanttag', tenant_tags)} --console
+      #{catch_powershell_error('Registering Tentacle')}
+    EOH
+    # This is sort of a hack, you need to specify the config_path on register if it is not default
+    # The other option is to read the registry key but the helpers are not available in 12.4.1
+    not_if { tentacle_exists?(server, api_key, tentacle_thumbprint(config_path)) }
+    notifies :restart, "windows_service[#{service_name}]", :delayed
+  end
+
+  windows_service service_name do
+    action :nothing
+  end
+end
+
+action :remove do
+  service_name = service_name(instance)
+
+  powershell_script "remove-tentacle-instance-#{instance}" do
+    action :run
+    cwd tentacle_install_location
+    code <<-EOH
+      .\\Tentacle.exe service --instance "#{instance}" --stop --uninstall --console
+      #{catch_powershell_error('Uninstalling tentacle service')}
+      .\\Tentacle.exe delete-instance --instance "#{instance}" --console
+      #{catch_powershell_error('Deleting instance from the server')}
+    EOH
+    only_if { ::Win32::Service.exists?(service_name) && ::File.exist?(tentacle_install_location) }
+  end
+
+  file config_path do
+    action :delete
+  end
+end
+
+action :uninstall do
```

Comment:
Made this attribute not remove the tentacle before uninstalling.  The main reason for this was because if we uninstall the service it would only remove one tentacle instead of multiples so it seems misrepresented